### DISCUSSION TITLE
RPM 4.20.0 build issue

### DISCUSSION THREAD

**Discussion by YellowJacketLinux** [rpm-software-management/rpm#3387] [General]
Hi, I'm having trouble building RPM 4.20.0 on an LFS (Linux From Scratch) 12.2 SystemD system.
I have an LFS 11.3 (SysV Init) system RPM bootstrapped with RPM 4.18.1 and in my new LFS 12.2 system I can get RPM 4.18.2 to build but I don't want to start with an EOL series.

The issue I am having (after figuring out I was missing "VERSION_ID" in my /etc/os-release file and fixing that) is that CMake is looking for either `podman` or `docker`. The actual error:

    CMake Error at tests/CMakeLists.txt:98 (find_program):
      Could not find PODMAN using the following names: podman, docker

Neither was mentioned as a build requirement in the INSTALL file (nor can I comprehend why they would be needed) and I can not seem to figure out what option I need to disable so CMake does not look for them.

This is my CMake command:

    cmake -DWITH_SELINUX=OFF -DWITH_SEQUOIA=OFF ..

Any suggestions?

(off-topic but I disabled sequoia because I have a strong dislike for Rust, it seems some thing break if you update and other things don't build if you don't update it, so on my LFS systems I am trying to avoid anything and everything that builds with Rust as long as I can)

**Comment by dmnks** [comment]
Podman/Docker is only needed for running the test-suite. You can disable it with `-DENABLE_TESTSUITE=OFF`.

The `VERSION_ID` field in `/etc/os-release` is also a test-suite specific thing and not needed when you disable the test-suite :)

Hope that helps!

**Reply by dmnks** [reply]
I should also add: While the test-suite is enabled by default in cmake, it is only really needed for development purposes. If you're just building RPM from source, you can safely disable it.

**Reply by YellowJacketLinux** [reply]
Thank you!

**Comment by dmnks** [comment]
This actually inspired me to file https://github.com/rpm-software-management/rpm/issues/3388. We always defaulted to the test-suite enabled (even with autoconf & fakechroot previously) but it might be a good idea to rethink that.

**Reply by YellowJacketLinux** [reply]
When I read the INSTALL file a little more carefully it did become obvious, as it names those tools. I had previously just skimmed the test suite part because I wasn't planning to run the test suite.

For what it's worth, the new RPM is working exactly as expected. I'm about twenty packages into bootstrapping my LFS build, only issue so far was completely my fault (kernel panic after RPM bootstrapping glibc - which doesn't automatically set executable bit on its libraries via make install...easy to fix fortunately). Anyway the new release is working well on a very bare-bones system.

**Reply by dmnks** [reply]
Thanks for the update, it's good to know!
